### ISSUE TITLE
Failed to load steamui.so: undefined symbol: va_fool_postp

### ISSUE THREAD

**Issue by stuarthayhurst** [ValveSoftware/steam-for-linux#8673]
#### Your system information

* Steam client version (build number or date): 1.0.0.74
* Distribution (e.g. Ubuntu): Debian Sid
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
On launch, `steamui.so` fails to load:
`Failed to load steamui.so - dlerror(): /home/ratus5/.steam/debian-installation/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libva-x11.so.2: undefined symbol: va_fool_postp`
I'm getting the issue after updating from mesa 22.1.0 to 22.1.3

#### Steps for reproducing this issue:
1. Attempt to launch Steam

**Comment by kisak-valve** [comment]
Hello @stuarthayhurst, while it's not clear to me why you've encountered this issue, you might be able to workaround this issue by installing the system variant of this library with something like `sudo apt install libva-x11-2:i386`.

**Comment by stuarthayhurst** [comment]
Well that's got it running, thanks :D
Was that just a missing dependency then, or an issue with the installed library?

**Comment by KirillNapupnik** [comment]
Hello, I can confirm the same error. Right after updating to **Libva 2.15.0** the error appeared. If you read the change log (https://github.com/intel/libva/releases/tag/2.15.0), it seems _**va_fool support**_ has been deprecated. I am no expert by any means, but I would think that caused the issue for people who have recently updated their Libva to the latest version.

**Comment by Spacefish** [comment]
i had the same issue, i am running Ubuntu 22.10 (under development currently)..

Failed to load steamui.so - dlerror(): /home/MYUSERNAME/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libva-x11.so.2: undefined symbol: va_fool_postp

strange as libva-x11.so.2 is under steams control / delivered by steam itself.. maybe va_fool_postp is only called if steam believes the GPU supports it? So it was not caught during testing?

**Comment by ahcox** [comment]
Same error on Ubuntu 22.04 here, using the bleeding edge Mesa PPA after an `update to: 22.2~git2207061133.6ab2e1~oibaf~j`

The same fix given above gets steam working again:

    sudo apt install libva-x11-2:i386

**Comment by Donno191** [comment]
Same issue but cannot get sudo apt install libva-x11-2 installed

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 libva-x11-2:i386 : Depends: libva2:i386 (< 2.14.0.1) but 2.15.0-1~oibaf~j is to be installed
E: Unable to correct problems, you have held broken packages.

**Comment by Donno191** [comment]
when i want to upgrade : sudo apt upgrade
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
The following packages have been kept back:
  libegl-mesa0:i386 libgbm1:i386 libgl1-mesa-dri:i386 libglapi-mesa:i386 libglx-mesa0:i386 libosmesa6:i386 mesa-va-drivers:i386 mesa-vulkan-drivers:i386
0 upgraded, 0 newly installed, 0 to remove and 8 not upgraded.

**Comment by JoneKone** [comment]
well updated Ubuntu and Boom the same error

```
Failed to load steamui.so - dlerror(): /home/xxx/.steam/debian-installation/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libva-x11.so.2: undefined symbol: va_fool_postp```

**Comment by cloudishBenne** [comment]
Indeed, it was caused by oibaf ppa.
Solution:
You do not need to remove the ppa, i have solved it without!
`sudo apt install --reinstall libva2:amd64=2.14.0-1 libva2:i386=2.14.0-1 libva-x11-2=2.14.0-1 libva-wayland2=2.14.0-1 libva-drm2=2.14
.0-1`

**Comment by smcv** [comment]
> strange as libva-x11.so.2 is under steams control / delivered by steam itself

Not entirely. The problem is that libva.so.2 and libva-x11.so.2 are meant to match, so the problematic situation here is if you have libva.so.2 on the host system (outside Steam's control), but not libva-x11.so.2. Steam needs to use whichever version of libva.so.2 is newer (choosing an older version could *also* cause crashes), and that will usually be the version from the host system - but if that doesn't match the libva-x11.so.2 provided with Steam, then you can get this crash.

Distributors can avoid this particular crash by making sure that their Steam packages pull in a system copy of both libva.so.2 and libva-x11.so.2 (at least 32-bit, but preferably 64-bit as well).

For users of Valve's `steam-launcher` package, a future update is likely to add a dependency on `libva-x11-2:i386`.

For users of Debian's `steam` package, version 1:1.0.0.74-2 fixes this.

For users of Ubuntu's `steam` package, I've opened https://bugs.launchpad.net/ubuntu/+source/steam/+bug/1981355.

**Comment by joshcangit** [comment]
> Hello @stuarthayhurst, while it's not clear to me why you've encountered this issue, you might be able to workaround this issue by installing the system variant of this library with something like `sudo apt install libva-x11-2:i386`.

Yes, this actually fixed it. Not sure why was this not included in `steam-libs-i386`.

**Comment by mxdpeep** [comment]
> Hello @stuarthayhurst, while it's not clear to me why you've encountered this issue, you might be able to workaround this issue by installing the system variant of this library with something like `sudo apt install libva-x11-2:i386`.

**worked for me too**

this error occurred right after installing ffmpeg4 and ffmpeg5 from it's PPA

**Comment by smcv** [comment]
> I think it should included as dependency of steam-libs-i386.
> Since that package recommends libva2:i386, make sense to add libva-x11-2:i386 to that as well.

That was done in 1.0.0.75, which was shipped as a beta in July and re-released as stable in September (the "future update" that I mentioned in my previous comment was exactly the change you suggest here).

The remaining problems with this seem to be from people who have added PPAs that might be providing `libva2` and `libva-x11-2` for amd64 but not for i386. PPAs are unofficial and not a supported part of the main OS that you're running.

Debian/Ubuntu multiarch relies on identical versions being available for each architecture, and the Debian/Ubuntu packages for Steam (both Valve's packages and the distributions' packages) heavily rely on multiarch, so unfortunately it's easy for a PPA to accidentally break the ability to install a library for both architectures at the same time.

**Comment by ssrublev** [comment]
> Same error on Ubuntu 22.04 here, using the bleeding edge Mesa PPA after an update to: `22.2~git2207061133.6ab2e1~oibaf~j`
> 
> The same fix given above gets steam working again:
> 
> ```
> sudo apt install libva-x11-2:i386
> ```

Worked on 22.10, thank you!

**Comment by S0litaire** [comment]
Encountered the same issue when updating Ubuntu 22.04 from AMD based GPU to Intel based GPU using Intel's instructions to install Intel GPU.

Installing libva-x11-2:i386 fixed the issue.

**Comment by joshchernoff** [comment]
I just recently updated all my libs locally, Ubuntu 22.4 AMD64,  Newest Mesa ect. like other libva-x11-2:i386 fixed issues.

**Comment by ski0090** [comment]
i had the same problem.
I solved it by directly build-installing libva with a new release tag.
https://github.com/intel/libva
